livekit-agents/livekit/agents/pipeline/io.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import annotations
+
+import asyncio
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import (
+    AsyncIterable,
+    Awaitable,
+    Callable,
+    Literal,
+    Optional,
+    Union,
+)
+
+from livekit import rtc
+
+from .. import llm, stt
+
+STTNode = Callable[
+    [AsyncIterable[rtc.AudioFrame]],
+    Union[Awaitable[Optional[AsyncIterable[stt.SpeechEvent]]]],
```

Comment:
the Union here isn't doing anything right?

it'd be great to provide example of what are valid output types for these